mailpoet/assets/js/src/newsletters/send/filter-segment.tsx

```diff
@@ -164,9 +164,9 @@ export function FilterSegment({
                 className="mailpoet-link"
                 key={i}
                 rel="noopener noreferrer"
-                onClick={(event) => {
+                onClick={async (event) => {
                   event.preventDefault();
-                  context.saveDraftNewsletter(() => {
+                  await context.saveDraftNewsletter().then(() => {
```

Comment:
The only tiny nitpick is I'd personally convert `await something().then(...)` to:

```ts
await something();
// here do the "then"
```

Reply:
I always appreciate "nitpick"s! I modified the commit so now 2a1eff63eb56a4419d8f63ec0352410086043f82 has it like you describe.